lib/assessment.js

```diff
@@ -4,44 +4,111 @@
 // Write a function that takes zero or more arguments
 // and returns the minimum from that set
 // assume that all the argument are numbers
-var min = function min() {
+// var min = function min([1, 3, 4, 2]) {
+
+//   var findMin = 0;
+
+//   for(var i = 0; i < min.length; i++) {
+//     return Math.min.apply();
+//   };
+
+// };
+
+// I am not sure how to write this correctly, but I would use Math.min to find the minimum number in an array.
+
+
+// correct answer:
+var min = function(){
+  var result = arg [0]; // first spot in the array - this is where to store the result
+  for(var index; index < arguments.length; index++); // iterate over the array
+    if(arguments[index] < result);
+      result = arguments[index];
+}
+return result; // have to see everything in the result, so return has to be outside the function
+
+
 
-};
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
-var sum = function sum() {
+// var sum = function sum([1, 2, 3, 4]) {
+
+//   var sumTotal = 0;
+//   for(var i = 0; i < sum.length; i++); {
+//     sumTotal += sum[i];
+//   }
+
+//   return sumTotal;
+
+// };
+
+// correct answer:
+var sum = function sum(){
+  var result = 0;
+  for (var index = 0; index < args.length; index ++);
+    result += arg[index];
+}
+return result;
 
-};
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
-var newArray = function newArray(length, defaultValue) {
+// var newArray = function newArray(length, defaultValue) {
 
+// };
+
+//corrected answer:
+var newArray = function newArray(length,defaultValue){
+  var result = [];
+  // for loop is to do something a certain number of  times
+  for (var i = 0; i < length; i++) { // length is the size of the new array
+    result.push(defValue);
```

Comment:
This statement is within the body of the `for` loop; it will be run multiple times. Well done. However, no variable called `defValue` exists in this function's scope.